### PR TITLE
Fix Parsers overloads

### PR DESCRIPTION
**Pull request by quinnj**
Fixes tests on main branch due to recent changes in Parsers xparse dispatch.

### CODE REVIEW

File: src/InlineStrings.jl [hunk 1]
```diff
@@ -817,7 +820,7 @@ function Parsers.xparse(::Type{T}, source::Union{AbstractVector{UInt8}, IO}, pos
     if overflowed
         code |= Parsers.OVERFLOW
     end
-    return Parsers.Result{S}(code, res.tlen, x)
+    return Parsers.Result{T}(code, res.tlen, x)
```
Comment (nickrobinson251) [inline src/InlineStrings.jl:823]: is `S` now entirely ignored by this `xparse` method? would passing a `S != T` have behaved differently before?

**Review by nickrobinson251 — APPROVED** [review]
thanks!

I thought something like this would be the eventual best fix but i was too tired to work it out 😆 

Also 1 question

File: src/InlineStrings.jl (hunk 1 above)
Comment (quinnj) [reply src/InlineStrings.jl:823]: Yes, essentially we're ignoring S now. Previously, I think that would have broken things.

Comment (nickrobinson251) [reply src/InlineStrings.jl:823]: looks like we were actually relying on this https://github.com/JuliaData/CSV.jl/issues/1033

Comment (nickrobinson251) [reply src/InlineStrings.jl:823]: fix https://github.com/JuliaStrings/InlineStrings.jl/pull/50

**Pull request merged**